email-reader/src/test/java/no/sikt/nva/email/reader/handler/VerifyScopusEmailReceivedHandlerTest.java

```diff
@@ -0,0 +1,123 @@
+package no.sikt.nva.email.reader.handler;
+
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import com.amazonaws.services.lambda.runtime.events.ScheduledEvent;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import no.sikt.nva.email.reader.model.exception.NoScopusEmailsReceived;
+import no.sikt.nva.email.reader.util.EmailGenerator;
+import no.unit.nva.stubs.FakeContext;
+import nva.commons.core.ioutils.IoUtils;
+import org.apache.james.mime4j.MimeException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.invocation.InvocationOnMock;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.sync.ResponseTransformer;
+import software.amazon.awssdk.http.AbortableInputStream;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+public class VerifyScopusEmailReceivedHandlerTest {
+
+    private static final String OBJECT_KEY = "someObjectKey";
+    private final String BUCKET_NAME = "someBucketName";
+
+    private final FakeContext context = new FakeContext() {
+        @Override
+        public String getInvokedFunctionArn() {
+            return randomString();
+        }
+    };
+    private final ScheduledEvent SCHEDULED_EVENT = new ScheduledEvent();
+    private S3Client s3Client;
+    private VerifyScopusEmailReceivedHandler handler;
+
+    @BeforeEach
+    void init() {
+        //Because lastModifiedFlag in listObject response is needed, the FaceS3Client is not usable
```

Comment:
I think the setup for making the FakeS3Client (fix the typo) return the desired response would be worse than what you're doing here. I'm sure a "beautiful" solution is possible, but it would be a massive refactoring of FakeS3Client, which is a relatively simple thing right now.